src/ec/packages/ec-component-link/link.html.twig

```diff
@@ -68,25 +71,49 @@
 {# Print the result #}
 
 <a href="{{ _link.path }}" class="{{ _css_class }}"{{ _extra_attributes|raw }}>
-  {%- if _icon.name is not empty %}
-    {% if _link.icon_position == 'before' %}
-      {% include '../ec-component-icon/icon.html.twig' with {
-        icon: _icon,
-        extra_classes: 'ecl-link__icon'
-      } only %}
-      {{- '&nbsp;' }}
-    {%- endif -%}
+  {%- if _icons is defined and _link.icon_position == 'before' -%}
+    {%- for icon in _icons -%}
+      {%- if icon.extra_classes is defined and icon.extra_classes is not empty -%}
+        {%- set icon_extra_classes = icon.extra_classes ~ ' ' ~ icon_extra_classes -%}       
+      {%- endif -%}
+      {%- set _icon = _icon|merge(icon) -%}
+      {%- include '../ec-component-icon/icon.html.twig' with _icon|merge({
+        icon: _icon|default({}),
+        extra_classes: icon_extra_classes
+      }) only -%}
+    {%- endfor -%}
+    {{- '&nbsp;' -}}
+  {%- elseif _icon.name is not empty and _link.icon_position == 'before' -%}
+    {%- include '../ec-component-icon/icon.html.twig' with _icon|merge({
+        icon: _icon|default({}),
+        extra_classes: icon_extra_classes
+      }) only %}
+    {{- '&nbsp;' -}}
+  {%- endif -%}
+  {%- if _icons is defined or (_icon.name is defined and _icon.name is not empty) -%}
     <span class="ecl-link__label">{{ _link.label }}</span>
-    {%- if _link.icon_position == 'after' %}
-      {{- '&nbsp;' }}
-      {%- include '../ec-component-icon/icon.html.twig' with {
-        icon: _icon,
-        extra_classes: 'ecl-link__icon'
-      } only %}
-    {% endif %}
-  {% else %}
-    {{- _link.label }}
-  {%- endif %}
+  {%- endif -%}
+  {%- if _icons is defined and _link.icon_position == 'after' -%}
+    {{- '&nbsp;' -}}
+    {%- for _icon in _icons -%}
```

Comment:
It may be cleaner to put the markup of the icons on a variable once instead of duplicating the code

Reply:
Yeah, i actually think that we might totally remove the distinction between the single and the multiple icon in the logic, what i mean is that i could probably just put the single one in an array and then we would just have two for cycles (before and after) with one include of the link component each.